src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -420,7 +420,7 @@ private StorageEnvironmentOptions CreateStorageEnvironmentOptions(DocumentDataba
             var indexTempPath = configuration.TempPath?.Combine(name);
 
             var options = configuration.RunInMemory
-                ? StorageEnvironmentOptions.CreateMemoryOnly(indexPath.FullPath, indexTempPath?.FullPath,
+                ? StorageEnvironmentOptions.CreateMemoryOnly(indexPath.FullPath, indexTempPath?.FullPath ?? indexPath.FullPath,
```

Comment:
This will put all temporary files in the same dir as data, won't it? Maybe we can change it to:

`Path.Combine(indexPath.FullPath, "Temp")`